src/librustdoc/clean/mod.rs

```diff
@@ -2524,7 +2523,6 @@ pub enum TypeKind {
     Struct,
     Union,
     Trait,
-    Variant,
```

Comment:
Is this *"some variants that cannot be constructed"*?

Reply:
Yes, this and `Type::Unique` triggered the dead code lint.